### DISCUSSION TITLE
Distributed lock - and autoscaling

### DISCUSSION THREAD

**Discussion by belthaZornv** [kedacore/keda#2151] [Ideas]
Everything is working as it should - although we do have a bigger process outside of EKS that acquires a distributed lock, which stops the workers from processing an SQS queue.

I'm curious if there's a way to stop autoscaling temporarily - possibly reading or polling for `NumberOfMessagesReceived` metric (or using a warm up attribute on Keda side of things -> and then checking if anything is being processed since they spun up?) and scaling down if there seems to be no workers reading the queue?

Any ideas would be great!

**Comment by arschles** [comment]
@belthaZornv one solution to this is to create your own external autoscaler rather than use the built-in SQS autoscaler. since that can sometimes be a lot to do, another thing you can do is build an in-cluster process that effectively pauses scaling when the external process acquires the lock. There's work ongoing to make this pausing functionality easier in #944, but until then, you can set `spec.maxReplicaCount` to zero to scale everything down to zero.

**Reply by belthaZornv** [reply]
Yeah, those were some of the options on the table; one thing I was thinking of is having multiple triggers, if the order of the triggers is maintained - then you can say - first trigger is redis -> value -> key -> max replica = 0, second trigger would be sqs -> scale up as usual.

Although not sure if the multiple triggers are working in that manner as of current.
